src/dbxdeploy/package/DependencyBuilder.py

```diff
@@ -0,0 +1,84 @@
+import time
+from pathlib import Path
+from pathlib import PurePosixPath
+from logging import Logger
+from databricks_api import DatabricksAPI
+from dbxdeploy.package.PackageMetadata import PackageMetadata
+from dbxdeploy.package.RequirementsCreator import RequirementsCreator
+from dbxdeploy.dbfs.DbfsFileUploader import DbfsFileUploader
+from dbxdeploy.dbfs.DbfsFileDownloader import DbfsFileDownloader
+
+
+class DependencyBuilder:
+    def __init__(
+        self,
+        project_base_dir: Path,
+        logger: Logger,
+        cluster_id: str,
+        default_job_cluster_definition: dict,
+        requirements_creator: RequirementsCreator,
+        dbfs_file_uploader: DbfsFileUploader,
+        dbfs_file_downloader: DbfsFileDownloader,
+        dbx_api: DatabricksAPI,
+    ):
+        self.__project_base_dir = project_base_dir
+        self.__cluster_id = cluster_id
+        self.__default_job_cluster_definition = default_job_cluster_definition
+        self.__logger = logger
+        self.__requirements_creator = requirements_creator
+        self.__dbfs_file_uploader = dbfs_file_uploader
+        self.__dbfs_file_downloader = dbfs_file_downloader
+        self.__dbx_api = dbx_api
+
+    def build(self, base_path: Path, package_metadata: PackageMetadata):
+        requirements = self.__requirements_creator.export_to_string(base_path, only_dependencies=True).splitlines()
+        dbfs_build_dir = PurePosixPath(f"dbfs:/tmp/DependencyBuild/{package_metadata.get_job_run_name()}")
+        dbfs_script_path = dbfs_build_dir.joinpath("download_dependencies.py")
+        dbfs_dependencies_dir = dbfs_build_dir.joinpath("dependencies")
+        local_dependencies_dir = base_path.joinpath("dependencies")
+        unix_dependencies_dir = "/dbfs/" + dbfs_dependencies_dir.as_posix().lstrip("dbfs:/")
+
+        script = (
+            f"import sys, subprocess\n"
+            f"requirements = {requirements}\n"
+            f'subprocess.run([sys.executable, "-m", "pip", "wheel", "-w", "{unix_dependencies_dir}", *requirements], capture_output=False, encoding="utf-8")\n'
+        ).encode("utf-8")
+
+        self.__logger.info("Uploading build script to dbfs...")
+        self.__dbfs_file_uploader.upload(script, str(dbfs_script_path))
+        self.__logger.info("Build Script uploaded")
+
+        self.__logger.info("Running remote build on Databricks cluster...")
+        self.__submit_python_script_on_job_cluster(dbfs_script_path, package_metadata, wait=True)
+        self.__logger.info("Build done")
+
+        self.__logger.info("Downloading built packages from dbfs...")
+        self.__dbfs_file_downloader.download_directory(dbfs_dependencies_dir, local_dependencies_dir, overwrite=True)
+        self.__logger.info("Download done")
```

Comment:
Download done, set dbxdeploy.target.package.offline_install: True in config and deploy